third-party/lz4/CMakeLists.txt

```diff
@@ -1,2 +1,29 @@
-add_definitions(-fPIC)
-add_library(lz4 STATIC src/lib/lz4.c src/lib/lz4.h src/lib/lz4hc.c src/lib/lz4hc.h)
+add_library(lz4 INTERFACE)
+
+find_package(LZ4)  # also checks minimum required version
+
+if (LZ4_FOUND)
+  target_include_directories(lz4 INTERFACE ${LZ4_INCLUDE_DIR})
+  target_link_libraries(lz4 INTERFACE ${LZ4_LIBRARY})
+
+else (LZ4_FOUND)
+  cmake_minimum_required(VERSION 2.8.0)
+  include(ExternalProject)
+
+  ExternalProject_Add(
+    lz4Build
+    URL "https://github.com/lz4/lz4/archive/v1.9.2.tar.gz"
+    URL_HASH SHA256=658ba6191fa44c92280d4aa2c271b0f4fbc0e34d249578dd05e50e76d0e5efcc
+    BUILD_IN_SOURCE true
```

Comment:
Sorry, now that INSTALL_DIR is being used, can you try turning this off?

Reply:
That creates an empty build directory and runs `make` in it. I think an out-of-source build requires a cmake-style configuration step that creates makefiles in the build directory.

But this should be fine, because "in source" here means inside the downloaded source, which itself is in HHVM's build directory.